### PR TITLE
Use proper parameter arithmetic in rewriters

### PR DESCRIPTION
**Pull request by notmgsk**
Pending test fixtures

### CODE REVIEW

File: src/compressor/rewriting-rules.lisp [hunk 1]
```diff
@@ -197,7 +197,7 @@
     ((x ("PISWAP" (theta) p q)
         :where (and (typep theta 'double-float)
                     (double= theta (* pi (floor theta pi))))))
-  (list (build-gate "PISWAP" `(,(- theta pi)) p q)
+  (list (build-gate "PISWAP" `(,(param-+ theta pi)) p q)
```
Comment (appleby) [inline src/compressor/rewriting-rules.lisp]: I assume switching from - to + here doesn't matter because theta + pi is equivalent to theta - pi?

Comment (ecpeterson) [reply src/compressor/rewriting-rules.lisp]: Good eye. We do need a `+` here: the `PISWAP` parameter is only 4pi-periodic.

File: tests/translator-tests.lisp [hunk 2]
```diff
@@ -90,76 +101,126 @@
...
+                                                  (gethash param parameter-assignments))
+                                                 (t
+                                                  (error 'test-case-too-complicated
+                                                         :reason "I don't know how to generate this parameter."))))
+                                             (quil::gate-binding-parameters binding))))
+                           (qubits (mapcar (lambda (qubit)
+                                             (cond
+                                               ((numberp qubit)
+                                                qubit)
+                                               ((eql qubit (intern "_"))
```
Comment (stylewarning) [inline tests/translator-tests.lisp]: no need to intern; just use `'_`

File: tests/translator-tests.lisp [hunk 3]
```diff
@@ -90,76 +101,126 @@
...
+                                   ((numberp qubit)
+                                    (push qubit output-qubits))
+                                   ((null qubit)
+                                    (push (random-with-exceptions qubit-bound
+                                                                  (append output-qubits occupied-qubits))
+                                          output-qubits)))))))
+                      (cond
+                        ((symbolp (quil::gate-binding-operator binding))
+                         (apply #'quil::anon-gate "ANONYMOUS-INPUT"
+                                (quil::random-special-unitary (ash 1 (length qubits)))
```
Comment (stylewarning) [inline tests/translator-tests.lisp]: prefer `(expt 2 (length qubits))`

**Comment by stylewarning** [comment]
(I added nitpicks because it's not ready-4-merge, according to @notmgsk)

**Pull request merged**